test/libsinsp_e2e/container/container_cri.cpp

```diff
@@ -0,0 +1,547 @@
+	                   bool extra_queries = true);
+
+	void fake_cri_test_timing(const std::string& pb_prefix,
+	                          const std::string& delay_arg,
+	                          const std::string& runtime,
+	                          float docker_delay,
+	                          bool async,
+	                          const exp_container_event_info& exp_info,
+	                          uint64_t container_engine_mask = 0,
+	                          int64_t test_duration = 10);
+};
+
+TEST_F(container_cri, fake_cri_no_server) {
+	std::atomic<bool> done(false);
+
+	event_filter_t filter = [&](sinsp_evt* evt) {
+		// we never get the PPME_CONTAINER_JSON_E event if the lookup fails
+		sinsp_threadinfo* tinfo = evt->get_tinfo();
+		if(tinfo) {
+			return tinfo->m_exe == "/bin/echo" && !tinfo->m_container_id.empty();
+		}
+
+		return false;
+	};
+
+	run_callback_t test = [&](sinsp* inspector) {
+		subprocess handle(LIBSINSP_TEST_PATH "/test_helper", {"cri_container_echo"});
+		handle.in() << "\n";
+		handle.wait();
+	};
+
+	captured_event_callback_t callback = [&](const callback_param& param) {
+		sinsp_threadinfo* tinfo = param.m_evt->get_tinfo();
+		EXPECT_TRUE(tinfo != NULL);
+
+		EXPECT_EQ(cri_container_id, tinfo->m_container_id);
+
+		const auto container_info =
+		        param.m_inspector->m_container_manager.get_container(tinfo->m_container_id);
+
+		// This can either be null or a container with incomplete metadata
+		EXPECT_TRUE(
+		        (container_info == nullptr ||
+		         container_info->get_lookup_status() != sinsp_container_lookup::state::SUCCESSFUL));
+
+		done = true;
+	};
+
+	before_capture_t setup = [&](sinsp* inspector) {
+		inspector->set_cri_socket_path(fake_cri_socket);
+	};
+
+	EXPECT_NO_FATAL_FAILURE({ event_capture::run(test, callback, filter, setup); });
+	EXPECT_TRUE(done);
+}
+
+void container_cri::fake_cri_test(
+        const std::string& pb_prefix,
+        const std::string& runtime,
+        const std::function<void(const callback_param& param, std::atomic<bool>& done)>& callback,
+        bool extra_queries) {
+	std::atomic<bool> done(false);
+	unlink(fake_cri_socket.c_str());
+	subprocess fake_cri_handle(LIBSINSP_TEST_PATH "/fake_cri/fake_cri",
+	                           {"unix://" + fake_cri_socket, pb_prefix, runtime});
+	pid_t fake_cri_pid = fake_cri_handle.get_pid();
+
+	auto start_time = time(NULL);
+
+	event_filter_t filter = [&](sinsp_evt* evt) {
+		return evt->get_type() == PPME_CONTAINER_JSON_E ||
+		       evt->get_type() == PPME_CONTAINER_JSON_2_E;
+	};
+
+	run_callback_t test = [&](sinsp* inspector) {
+		subprocess handle(LIBSINSP_TEST_PATH "/test_helper", {"cri_container_echo"});
+		handle.in() << "\n";
+		handle.wait();
+		while(!done && time(NULL) < start_time + 10) {
+			usleep(100000);
+		}
+	};
+
+	captured_event_callback_t cri_callback = [&](const callback_param& param) {
+		callback(param, done);
+	};
+
+	before_capture_t setup = [&](sinsp* inspector) {
```

Comment:
We need to reset these to the correct values after the test, at least the docker_socket_path, since it is static.
See for example `TEST_F(sys_call_test, container_docker_bad_socket)`:
```
after_capture_t cleanup = [&](sinsp* inspector) {
		inspector->set_docker_socket_path("/var/run/docker.sock");
	};
```

Also, if the `inspector->set_docker_socket_path("");` is needed to "disable" docker, we can just use `inspector->set_container_engine_mask(1 << CT_CRI)` to only enable cri.

Reply:
Uh, didn't know that api! I would rather add another test using that instead of changing that one: that way we'll be certain of the robustness :)

Reply:
Thanks bobo!

Reply:
I saw that API is indeed used in that source file though :D